tests/test_routing.py

```diff
@@ -43,3 +45,23 @@ def test_request_with_trailing_slash_not_redirected(
     """
     response = test_app.get(invalid_route)
     assert response.status_code == 404
+
+
+@pytest.mark.parametrize(
+    "test_root_path,expected_status_code",
+    [("", 200), ("/api/v1", 200), ("/wrongroot", 404)],
+)
+def test_docs_work_using_defined_root_path(
+    test_app, test_root_path, expected_status_code, monkeypatch
+):
+    monkeypatch.setattr(app, "root_path", "/api/v1")
+    docs_response = test_app.get(
+        f"{test_root_path}/docs", follow_redirects=False
+    )
+    # When the root path is not set correctly, the docs can break due to failure to fetch openapi.json
```

Comment:
What is the relevance of this info in the test? Is this an explanation for why you expect a 404 for `/wrongroot`?

Reply:
Yes, as well as the motivation for why we test reachability of not just `.../docs` but also of `.../openapi.json` as a sanity check! Moved this note to the docstring instead for context.